src/error.rs

```diff
@@ -115,8 +116,49 @@ pub enum Kind {
     WrongFileNamePattern(PathBuf),
     /// Conversion error
     Conversion(String),
-    /// Record validation errors, eg. magic byte check
-    Validation(String),
+    /// Validation errors, eg. magic byte check
+    Validation {
+        /// Describes what check failed.
+        param: ValidationParam,
+        /// Description of an error cause.
+        cause: String,
+    },
+
     /// Other error
     Other,
 }
+
+/// Variants of validation errors.
+#[derive(Debug, Clone, PartialEq)]
+pub enum ValidationParam {
```

Comment:
Maybe `ValidationErrorKind` is more descriptive name? And also `param` -> `kind` in field name